includes/class-blocks.php

```diff
@@ -469,4 +432,37 @@ function ( $paragraph ) {
 
 		return $data;
 	}
+
+	/**
+	 * Add directions to the button.
+	 *
+	 * @param string $content    The block content.
+	 * @param array  $attributes The block attributes.
+	 *
+	 * @return string The updated content.
+	 */
+	public static function add_directions_to_button( $content, $attributes ) {
+		$tags = new \WP_HTML_Tag_Processor( $content );
+
+		$button_class = '';
+		if ( isset( $attributes['buttonSize'] ) && 'small' === $attributes['buttonSize'] ) {
+			$button_class = 'is-small';
+		} elseif ( isset( $attributes['buttonSize'] ) && 'compact' === $attributes['buttonSize'] ) {
+			$button_class = 'is-compact';
+		}
```

Comment:
I'm leaning towards removing this functionality and just keep customization options to whatever the core button block offers. What do you think?

Reply:
I think it is important to have a "button" only mode, but let's remove the sizing for now.

Reply:
Fixed in f86bc55b4a794930d687c3520c05de9515b41ae8